src/schedule/useSchedule.ts

```diff
@@ -130,6 +138,93 @@ const useSchedule = create<IUseSchedule>(
           isLoading: false,
         }));
       },
+      validateSchedule: async (
+        scheduleItem: IScheduleItem,
+        contract: IContract,
+        provider: IProvider,
+        myAccountAddress: string
+      ): Promise<IScheduleFormDialogAlert[]> => {
+        set(() => ({
+          isLoading: true,
+        }));
+
+        const selectedPlan = provider.plans[+scheduleItem.providerPlanIndex];
+
+        const result: IScheduleFormDialogAlert[] = [];
+
+        // validate purchased execution
+        const hasAnExecutionLeft = selectedPlan.remainingExecutions?.gt(0);
+        if (!hasAnExecutionLeft) {
+          result.push({
+            message: "You don't have executions left",
+            severity: "error",
+            actionLabel: "Purchase more",
+            actionLink: "/store",
+          });
+        }
+
+        const encodedFunctionCall = new utils.Interface(
+          contract.ABI
+        ).encodeFunctionData(
+          scheduleItem.contractMethod,
+          scheduleItem.contractFields
+        );
+
+        // if gasEstimation is undefined warn the user that the execution might fail
+        const estimatedGas = await provider.contractInstance.estimateGas(
+          contract.address,
+          encodedFunctionCall
+        );
+        if (!estimatedGas) {
+          result.push({
+            message:
+              "We couldn't estimate the gas for the execution you want to schedule. " +
+              "Hint: Make sure that all the parameters are correct and " +
+              "that you will comply the execution requirements at the time of its execution.",
+            severity: "warning",
+          });
+        }
+
+        // estimatedGas is lower than the gas limit for the selected plan
+        const isInsidePlanGasLimit = selectedPlan.gasLimit.gte(
+          estimatedGas ?? 0
+        );
+        if (estimatedGas && !isInsidePlanGasLimit) {
+          result.push({
+            message:
+              `The selected plan has a gas limit of ${formatBigNumber(
+                selectedPlan.gasLimit,
+                0
+              )} which is lower that ` +
+              `our estimation for this execution (${formatBigNumber(
+                estimatedGas,
+                0
+              )}).`,
+            severity: "warning",
+          });
+        }
+
+        // validate minimum date/time
+        const minimumDate = addSeconds(
+          new Date(),
+          environment.MINIMUM_TIME_BEFORE_EXECUTION
+        );
+
+        if (parseISO(scheduleItem.executeAt) <= minimumDate) {
```

Comment:
We should add this validaton to the SDK. We can use this method https://github.com/rsksmart/rif-scheduler-contracts/blob/develop/contracts/RIFScheduler.sol#L50

Let's rise an issue in SDK repo